contrib/reactphp/src/Toolshed/Toolshed.php

```diff
@@ -0,0 +1,148 @@
+<?php namespace Cjdns\Toolshed;
+
+use Cjdns\Bencode\Bencode;
+
+class Toolshed {
+
+    static function logger($mesg=null) {
+        if ($mesg) {
+            echo Toolshed::logo() . $mesg . PHP_EOL;
+        } else {
+            return;
+        }
+    }
+
+    static function logo() {
+        return '[ᴄᴊᴅɴꜱ] ';
+    }
+
+    static function logger_http($mesg=null, $req=null) {
+        if (!$mesg) {
+            return;
+        } else if (!$req) {
+            echo Toolshed::phlogo() . $mesg . PHP_EOL;
+        }
+
+
+    }
+
+    static function phlogo() {
+        return '[Pʜʟᴜɪᴅ-ʜᴛᴛp] ';
+    }
+
+    static function ramusage($cfg, $verbose=false) {
+
+        /* $kmem(int) returns formated bytes */
+        $kmem = function($b) { return number_format($b / 1024). 'KiB'; };
+
+        /* Bytes in use for server.php */
+        $bytesphp = $kmem(memory_get_usage(true));
+
+        /* Get cjdns bytes from Admin Socket */
+        $data = \Cjdns\Api\Api::Allocator_bytesAllocated();
+        $Socket = new \Cjdns\Admin\Socket($cfg);
+        $Socket->verbose = false;
+        $Socket->authput($data);
+
+        /* Bytes in use for cjdns */
+        $bytesalloc = $kmem(\Cjdns\Api\Api::decode($Socket->message)['bytes']);
+
+        if ($verbose) {
+            Toolshed::logger("Memory ReactPHP: " . $bytesphp . ", cjdns: " . $bytesalloc);
+        } else {
+            return "Memory ReactPHP: " . $bytesphp . ", cjdns: " . $bytesalloc;
+        }
+    }
+
+    /* */
+    static function publictoip6($pubkey = false) {
+        if (!$pubkey) { return false; }
+        $bin = '../../publictoip6 ';
+        $h   = popen($bin . $pubkey, 'r');
```

Comment:
Assuming `$pubkey` can be controlled by a user, we have an issue here (didn't verify)